src/http/routing.rs

```diff
@@ -902,11 +902,23 @@ impl Route {
         format!(api!("/webhooks/{}/{}"), webhook_id, token)
     }
 
-    pub fn webhook_with_token_optioned<D>(webhook_id: u64, token: D, wait: bool) -> String
+    pub fn webhook_with_token_optioned<D>(
+        webhook_id: u64,
+        token: D,
+        wait: bool,
+        thread_id: Option<u64>,
+    ) -> String
     where
         D: Display,
     {
-        format!(api!("/webhooks/{}/{}?wait={}"), webhook_id, token, wait)
+        if let Some(thread_id) = thread_id {
+            format!(
+                api!("/webhooks/{}/{}?wait={}&thread_id={}"),
+                webhook_id, token, wait, thread_id
+            )
+        } else {
+            format!(api!("/webhooks/{}/{}?wait={}"), webhook_id, token, wait)
+        }
```

Comment:
```suggestion
       let mut path = format!(api!("/webhooks/{}/{}?wait={}"),
                webhook_id, token, wait);

       if let Some(thread_id) = thread_id {
          let _ = write!(path, "&thread_id=", thread_id);
   }

path
        
```

(You'll need to reformat it)